cmd/clusterctl/client/cluster/mover.go

```diff
@@ -234,6 +236,19 @@ func newObjectMover(fromProxy Proxy, fromProviderInventory InventoryClient) *obj
 	return &objectMover{
 		fromProxy:             fromProxy,
 		fromProviderInventory: fromProviderInventory,
+		namespaceRefFieldsForKnownKinds: map[string][][]string{
+			"Cluster": {
+				{"spec", "controlPlaneRef", "namespace"},
+				{"spec", "infrastructureRef", "namespace"},
+			},
+			"KubeadmControlPlane": {
+				{"spec", "machineTemplate", "infrastructureRef", "namespace"},
+			},
+			"Machine": {
+				{"spec", "bootstrap", "configRef", "namespace"},
+				{"spec", "infrastructureRef", "namespace"},
+			},
+		},
```

Comment:
Strictly speaking, you don't need to consider the namespace of any "reference" fields. Although they are of the v1.ObjectReference go type, the namespace field is not used. https://github.com/kubernetes-sigs/cluster-api/issues/6539 describes the work in progress to use a different go type. However, while that work is not merged, it may be important to respect the namespace fields.

Reply:
thanks for linking to that upstream issue. 

but yeah i agree, until that can be merged we need to update the fields, cuz otherwise admisison webhook is failing the `Cluster` creation saying `.metadata.namespace != spec.someRef.namespace`